### DISCUSSION TITLE
Ftp not working

### DISCUSSION THREAD

**Discussion by TaNorbs** [pockethost/pockethost#347] [Q&A] [answered]
Hi,

I am trying to ftp into my instance, so i could drop some files into pb_hooks. Using FileZilla, and the ftp link on the instance page. The console says that this user exists, but when i put in my password, it errors with invalid username or password, even though i can log into pockethost without any hitch.

I tried resetting my password twice now, even tried it with some very basic password without any success. 

Thank you in advance!

**Comment by Ilyas-Codes** [comment]
I'm having the exact same issue, does pockethost have another way of connecting to my instance with ssh or something else other than ftp?

**Comment by benallfree** [comment]
Hi guys, I just tested with FileZilla and it is working. 

Please make sure you are using `ftp.sfo-1.pockethost.io` as the host.

If you are still having trouble, please DM me on Discord

**Comment by Ilyas-Codes** [comment]
oh thank you, so your method worked for me, I used the command "ftp ftp.sfo-1.pockethost.io" and provided the username and password in the command line. this method was successful. but, when I was trying to use the command provided in the Pockethost instance area, which is: "ftp ftp://ilyas.codes%40icloud.com@ftp.sfo-1.pockethost.io" it didn't work. just wanted to let you know about this. Thank you for the quick response and for making Pockethost.

**Reply by benallfree** [reply]
Ah, do you mean the example in the documentation is not working?

**Reply by benallfree** [reply]
I just tested it and it worked for me:

```bash
✗  ftp ftp://ben%40benallfree.com@ftp.sfo-1.pockethost.io
Connected to ftp.sfo-1.pockethost.io.
220 Ready
331 Username okay, awaiting password
Password:
230 User logged in, proceed
Remote system type is UNIX.
Using binary mode to transfer files.
200 Switch to "binary" transfer mode.
ftp> ls
229 EPSV OK (|||10084|)
150 File status okay; about to open data connection
drwxr-xr-x 1 1 1            0 Nov 05 11:06 harvest
drwxr-xr-x 1 1 1            0 Nov 08 07:04 harvesttest
drwxr-xr-x 1 1 1            0 Nov 05 11:06 able-salesmen
drwxr-xr-x 1 1 1            0 Nov 05 11:25 noisy-france
drwxr-xr-x 1 1 1            0 Nov 05 11:06 little-window
drwxr-xr-x 1 1 1            0 Nov 05 11:06 breezy-mechanic
drwxr-xr-x 1 1 1            0 Nov 05 11:06 blue-photographer
drwxr-xr-x 1 1 1            0 Nov 05 11:28 screeching-child
drwxr-xr-x 1 1 1            0 Nov 07 23:51 testers
drwxr-xr-x 1 1 1            0 Nov 05 11:06 ashamed-gigabyte
drwxr-xr-x 1 1 1            0 Nov 05 11:06 modern-answer
drwxr-xr-x 1 1 1            0 Nov 24 17:44 bitter-mothers
drwxr-xr-x 1 1 1            0 Nov 05 11:06 acoustic-telephone
drwxr-xr-x 1 1 1            0 Nov 07 23:52 victorious-salesclerk
drwxr-xr-x 1 1 1            0 Nov 15 01:32 eager-tailor
drwxr-xr-x 1 1 1            0 Nov 08 04:59 brave-restaurant
drwxr-xr-x 1 1 1            0 Nov 15 01:29 fat-cpu
drwxr-xr-x 1 1 1            0 Nov 15 01:26 bright-leather
drwxr-xr-x 1 1 1            0 Dec 01 18:33 receive-empty
226 Closing data connection
ftp>
```

**Reply by TaNorbs** [reply]
Same case for me as well. Using the longer command 'ftp ftp://NorbsMisc%40protonmail.com@ftp.sfo-1.pockethost.io' and then putting in my password does not work, but when connecting to 'ftp.sfo-1.pockethost.io' using FileZilla it does work, I could transfer the files with it.

Its weird tho, how on the CLI (Linux + zsh) I can successfully login, but when using the 'ls' command it says 'The requested action has been successfully completed' but then 'No connection established'.

**Reply by benallfree** [reply, marked as answer]
Can you please try https://github.com/pockethost/pockethost/discussions/345#discussioncomment-7734690

```bash
ftp "No*****c@protonmail.com"@ftp.sfo-1.pockethost.io
```

**Reply by TaNorbs** [reply]
Yes, this actually works!

Should I mark this as answer?

**Reply by benallfree** [reply]
Yes please! I'm going to update the code too.

**Reply by calebfaruki** [reply]
This doesn't work for me. Using tnftp on mac. It looks like my username gets truncated even if I wrap the email in quotes.

```
ftp -v "c***@proton.me"@ftp.sfo-1.pockethost.io
ftp: Can't lookup `proton.me@ftp.sfo-1.pockethost.io:ftp': nodename nor servname provided, or not known
```

It can't connect to the host when tested using curl as well:

```
curl -v -u "c***@proton.me:****" ftps://ftp.sfo-1.pockethost.io/ --ssl-reqd -k
*   Trying 147.182.196.168:990...
* connect to 147.182.196.168 port 990 failed: Operation timed out
* Failed to connect to ftp.sfo-1.pockethost.io port 990 after 75009 ms: Couldn't connect to server
* Closing connection 0
curl: (28) Failed to connect to ftp.sfo-1.pockethost.io port 990 after 75009 ms: Couldn't connect to server
```

**Reply by benallfree** [reply]
Following up for @calebfaruki, FTP is listening on port 21, not port 990. Curl defaults ftps to 990 but most clients default to 21 even for ftps. Dashboard instructions have been updated to be explicit about port 21.

**Comment by Ilyas-Codes** [comment]
it doesn't work for me for some reason, after inputting the password I get a 530 error:

ftp ftp://ilyas.codes%40icloud.com@ftp.sfo-1.pockethost.io
Connected to ftp.sfo-1.pockethost.io.
220 Ready
331 Username okay, awaiting password
Password: 
530 Invalid username or password
ftp: Login failed.
ftp>

**Reply by benallfree** [reply]
I wonder if your pw has some special characters that the CLI doesn't like?

**Reply by Ilyas-Codes** [reply]
don't think so, the password works fine when logging in with `ftp ftp.sfo-1.pockethost.io`

**Comment by Ilyas-Codes** [comment]
Hey, sorry for taking your time, it's my first time working with FTP and I'm stuck on this, when I try copying the pocketbase executable from my local computer over to the FTP server using this command: "put /Users/Ilyas/Desktop/pb/pocketbase" inside the top level of my pockethost instance directory, I get this error: "550 Accessing Ilyas is not allowed." am I doing this all wrong?

**Reply by benallfree** [reply]
That is correct, custom executables are not allowed at this time. Sorry.

**Reply by Ilyas-Codes** [reply]
I see, so basically I have to recreate my pocketbase collection and data in the pockethost instance right?

**Reply by benallfree** [reply]
PocketBase migrations go in `pb_migrations` and you can upload those. You just can't upload a custom `pocketbase` binary.

**Reply by Ilyas-Codes** [reply]
Yes, I figured it out lol, I was just having trouble working with the FTP command, very confusing at first. thank you!
